auth/id_token_from_impersonated_credentials.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+// [START auth_cloud_idtoken_impersonated_credentials]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/impersonate"
+	"google.golang.org/api/option"
+)
+
+// getIdTokenFromImpersonatedCredentials uses a service account (SA1) to impersonate as
+// another service account (SA2) and obtain id token for the impersonated account.
+// To obtain token for SA2, SA1 should have the "roles/iam.serviceAccountTokenCreator" permission on SA2.
+func getIdTokenFromImpersonatedCredentials(w io.Writer, scope, targetAudience, impersonatedServiceAccount string) error {
+	// scope := "https://www.googleapis.com/auth/cloud-platform"
+	// targetAudience := "http://www.example.com"
+	// impersonatedServiceAccount := "name@project.service.gserviceaccount.com"
+
+	ctx := context.Background()
+
+	// Construct the GoogleCredentials object which obtains the default configuration from your
+	// working environment.
+	credentials, err := google.FindDefaultCredentials(ctx, scope)
```

Comment:
Not creating this type and passing this has the same affect, can you remove this from the example?

Reply:
It will work, but the goal of the samples is to show how to explicitly set up the credentials
Examples from other languages: https://github.com/googleapis/google-auth-library-python/blob/main/samples/cloud-client/snippets/idtoken_from_impersonated_credentials.py#L47

Reply:
For these two can we add to the comment to say that this would be done implicitly or something to that nature? I think it is fine to be explicit. But if we don't say we are being verbose users are going to follow this pattern without knowing that there is a simpler way.

Reply:
I guess we have "authenticate explicit" example and it should be enough from my view.

Again, same for other languages (there are additional info): https://github.com/googleapis/google-auth-library-python/pull/1102/files#diff-96271f4091235622616d8e8fa7d45ddf6da62cd552544721dc10214a3562d57dR47

Reply:
That is true there is a sample like that but think about it from a user point of view. I don't think that sample and this one will be shown on the same page, right? If there are not they don't have the context of the other sample. I get that this is how things were done in other langs too, but I wonder if it is right to do so there as well.

Whenever possible I think we should have users rely on our auto-detection logic and only in exceptional cases be explicit, when required. I will approve this, but might be worth talking to the team about.